File: etamen-backend/app/Modules/Appointments/Infrastructure/Models/AppointmentStatusHistory.php
<?php

namespace App\Modules\Appointments\Infrastructure\Models;

use App\Models\User;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class AppointmentStatusHistory extends Model
{
    public const UPDATED_AT = null;

    protected $fillable = [
        'appointment_id',
        'from_status',
        'to_status',
        'actor_id',
        'reason',
        'metadata',
    ];

    protected function casts(): array
    {
        return [
            'metadata' => 'array',
        ];
    }

    public function appointment(): BelongsTo
    {
        return $this->belongsTo(Appointment::class);
    }

    public function actor(): BelongsTo
    {
        return $this->belongsTo(User::class, 'actor_id');
    }
}

File: etamen-flutter/lib/features/appointments/domain/entities/appointment_status_history.dart
class AppointmentStatusHistory {
  const AppointmentStatusHistory({
    required this.toStatus,
    this.fromStatus,
    this.reason,
    this.createdAt,
  });

  final String? fromStatus;
  final String toStatus;
  final String? reason;
  final DateTime? createdAt;
}
